torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4288,6 +4307,73 @@ def sample_inputs_narrow(op_info, device, dtype, requires_grad, **kwargs):
         yield sample
         yield SampleInput(sample.input, args=(sample.args[0], torch.tensor(sample.args[1]), sample.args[2]))
 
+def error_inputs_narrow_narrow_copy(op_info, device):
+    make_arg = partial(make_tensor, device=device, dtype=torch.float32)
+
+    # shape, args, error_type, error_regex
+    samples = [
+        # 0-dim
+        ((), (0, 0, 1), RuntimeError,
+         r"narrow\(\) cannot be applied to a 0-dim tensor\."),
+
+        # out of bounds dim is tested below
+        # out of bounds dim (negative)
+        ((L, S, M), (-4, 0, 0), IndexError,
+         r"Dimension out of range \(expected to be in range of \[-3, 2\], but got -4\)"),
+
+        # out of bounds start is tested below
+        # out of bounds start (negative), negative start means counting from the right
+        ((L, M, S), (1, -M - 1, 0), IndexError,
+         r"Dimension out of range \(expected to be in range of \[-10, 9\], but got -11\)"),
+
+        # out of bounds length
+        ((S, L, M), (2, 0, M + 1), RuntimeError,
+         r"start \(0\) \+ length \(11\) exceeds dimension size \(10\)\."),
+        # out of bounds length (negative)
+        ((M,), (0, 0, -1), RuntimeError,
+         r"start \(0\) \+ length \(-1\) exceeds dimension size \(10\)\."),
+    ]
+
+    # narrow_copy_dense_cpu_out in core has slightly different checks:
+    # https://github.com/pytorch/pytorch/issues/87019
+    # out of bounds dim
+    if op_info.name == 'narrow_copy' and device == 'cpu':
+        samples.append(
+            ((M, S, L), (3, 0, 0), RuntimeError,
+             r"Expected dim < static_cast<int64_t>\(self_sizes.size\(\)\) to be true, but got false\."))
+    else:
+        samples.append(
+            ((M, S, L), (3, 0, 0), IndexError,
+             r"Dimension out of range \(expected to be in range of \[-3, 2\], but got 3\)"))
+
+    # out of bounds start
+    if op_info.name == 'narrow_copy' and device == 'cpu':
+        samples.append(
+            ((L, M, S), (1, M + 1, 0), RuntimeError,
+             r"start \(11\) \+ length \(0\) exceeds dimension size \(10\)\."))
+    else:
+        samples.append(
+            ((L, M, S), (1, M + 1, 0), IndexError,
+             r"Dimension out of range \(expected to be in range of \[-10, 9\], but got 11\)"))
```

Comment:
The error is raised correctly. However, the error message could be improved because the allowed upper bound should be `10`. But this will only happen with `length=0`, which will return an empty result. So I'm not sure whether it's worth introducing additional code just to handle this case.

```py
In [10]: x
Out[10]: tensor([1, 2, 3, 4, 5, 6])

In [11]: x.size()
Out[11]: torch.Size([6])

In [12]: torch.narrow(x, 0, 6, 0)
Out[12]: tensor([], dtype=torch.int64)
```

Reply:
I'd say it's fine to leave it as it is.